### PR TITLE
Update Interoperability doc with array, map mapping

### PR DESCRIPTION
**Pull request by gabilang**
## Check List

- [ ] **Page Addition**
  - [ ] Add `permalink` to pages
  - [ ] If contains empty folder(s), Add front-matter `redirect_to:`

- [ ] **Page Rename**
  - [ ] Add front-matter `redirect_from`
  - [ ] Add front-matter `redirect_to:` (If applicable)

### CODE REVIEW

File: learn/calling-java-code-from-ballerina.md [hunk 1]
```diff
@@ -1223,7 +1223,79 @@ boolean | boolean |
 byte | byte, short, char, int, long, float, double | Widening conversion from byte -> short, char, int, long, float, double
 int | byte, char, short, int, long | Narrowing conversion when int -> byte, char, short and int
 float | byte, char, short, int, long, float, double | Narrowing conversion when float -> byte, char, short, int, long, float
+int[], float[], string[], byte[], boolean[] | io.ballerina.runtime.api.values.BArray |
+map | io.ballerina.runtime.api.values.BMap |
+
+#### Mapping Arrays and Map Types from Java to Ballerina Types
+Ballerina does not provide direct mapping of Java arrays and maps types to ballerina. In order to do mapping of arrays and map types from java to ballerina, you have to add the `ballerina-runtime` library to your java project and need to import relevant classes from the `ballerina-runtime` library. You can find the latest version of the library [here](https://maven.wso2.org/nexus/content/repositories/releases/org/ballerinalang/ballerina-runtime/).
```
[the remark this answers is not in the document]
Comment (riyafa) [reply learn/calling-java-code-from-ballerina.md:1230]: Wouldn't it be better to mention how to add runtime library as a dependency in maven and in gradle without only providing a link to where we can download the jar?

File: learn/calling-java-code-from-ballerina.md [hunk 2]
```diff
@@ -1223,7 +1223,79 @@ boolean | boolean |
...
+##### Mapping of Arrays from Java to Ballerina
+In order to do mapping `arrays` from java to ballerina, you need to use `BArray` to enable array functionalities that support Ballerina.
```
Comment (riyafa) [inline learn/calling-java-code-from-ballerina.md:1232]:
```suggestion
To use Ballerina arrays in Java BArray interface has been provided. 
```

File: learn/calling-java-code-from-ballerina.md [hunk 3]
```diff
@@ -1223,7 +1223,79 @@ boolean | boolean |
...
+Following example illustrates how to write  to Ballerina.
```
Comment (riyafa) [inline learn/calling-java-code-from-ballerina.md:1233]:
```suggestion
Following example illustrates how to write a the Java intreop code that uses Ballerina arrays 
```

File: learn/calling-java-code-from-ballerina.md [hunk 4]
```diff
@@ -1223,7 +1223,79 @@ boolean | boolean |
...
+Java code that returns `BArray` type:
```
Comment (riyafa) [inline learn/calling-java-code-from-ballerina.md:1234]:
```suggestion
Java code that uses `BArray` type:
```

File: learn/calling-java-code-from-ballerina.md [hunk 5]
```diff
@@ -1223,7 +1223,79 @@ boolean | boolean |
...
+       for(long i = len - 1, j = 0; j < len/2; i--, j++) {
+           Object temp = arr.get(j);
+           arr.add(j, arr.get(i));
+           arr.add(i, temp);
+       }
+       return arr;
+   }
+}
+```
+```ballerina
```
Comment (riyafa) [inline learn/calling-java-code-from-ballerina.md:1250]:
```suggestion

Associated Ballerina code:
```ballerina
```

File: learn/calling-java-code-from-ballerina.md [hunk 6]
```diff
@@ -1223,7 +1223,79 @@ boolean | boolean |
 byte | byte, short, char, int, long, float, double | Widening conversion from byte -> short, char, int, long, float, double
 int | byte, char, short, int, long | Narrowing conversion when int -> byte, char, short and int
 float | byte, char, short, int, long, float, double | Narrowing conversion when float -> byte, char, short, int, long, float
+int[], float[], string[], byte[], boolean[] | io.ballerina.runtime.api.values.BArray |
+map | io.ballerina.runtime.api.values.BMap |
+
+#### Mapping Arrays and Map Types from Java to Ballerina Types
```
Comment (riyafa) [inline learn/calling-java-code-from-ballerina.md:1229]:
```suggestion
#### Using Ballerina arrays and maps in Java

File: learn/calling-java-code-from-ballerina.md [hunk 7]
```diff
@@ -1223,7 +1223,79 @@ boolean | boolean |
...
+Ballerina does not provide direct mapping of Java arrays and maps types to ballerina. In order to do mapping of arrays and map types from java to ballerina, you have to add the `ballerina-runtime` library to your java project and need to import relevant classes from the `ballerina-runtime` library. You can find the latest version of the library [here](https://maven.wso2.org/nexus/content/repositories/releases/org/ballerinalang/ballerina-runtime/).
+##### Mapping of Arrays from Java to Ballerina
```
Comment (riyafa) [inline learn/calling-java-code-from-ballerina.md:1231]:
```suggestion
##### Using Java arrays in Ballerina
```

File: learn/calling-java-code-from-ballerina.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (warunalakshitha) [reply learn/calling-java-code-from-ballerina.md:1230]: shall we add sample gradle config

File: learn/calling-java-code-from-ballerina.md (hunk 7 above)
Comment (warunalakshitha) [reply learn/calling-java-code-from-ballerina.md:1231]: or using ballerina values in Java

File: learn/calling-java-code-from-ballerina.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (gabilang) [reply learn/calling-java-code-from-ballerina.md:1231]: Sure, 
```
repositories {
   // Use WSO2's Nexus repository manager for resolving dependencies.
   maven {
       url = 'https://maven.wso2.org/nexus/content/repositories/releases/'
   }
}

dependencies {
   // Add ballerina-runtime:2.0.0-Preview8 as dependency.
   implementation 'org.ballerinalang:ballerina-runtime:2.0.0-Preview8'
}
```
Adding this is enough right?

**Pull request closed without merging**